pkg/provisioner/helpers.go

```diff
@@ -25,6 +25,19 @@ func makeObjectReference(claim *v1alpha1.ObjectBucketClaim) *corev1.ObjectRefere
         }
 }
 
+func makeOwnerReference(claim *v1alpha1.ObjectBucketClaim) metav1.OwnerReference {
+	blockOwnerDeletion := true
+	isController := true
+
+	return metav1.OwnerReference{
+		Kind: "ObjectBucketClaim",
```

Comment:
- missing ApiVersion, Kind should not be hardcoded

getting it programatically looks like

```
   	groupVersion, kind := claim.GetObjectKind().GroupVersionKind().ToAPIVersionAndKind()
        ...
            Kind: kind,
            ApiVersion: groupVersion,
        ...
```